### ISSUE TITLE
Nextflow error handling & BusyBox images

### ISSUE THREAD

**Issue by bosterholz** [metagenomics/metagenomics-tk#230]
Today I noticed a strange bug. CI run "CI #523" failed during the Plsdb test despite nothing being changed in the plasmid module. 
Error:
```
nxf-scratch-dir bibigrid-worker-1-3-6yqgmh3vqbon1hz:/vol/scratch/nxf.BxZTQHP2X9
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Database will be downloaded!
Connecting to openstack.cebitec.uni-bielefeld.de:8080 (129.70.51.3:8080)
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 80): 80
wget: error getting response: Connection reset by peer
MD5SUM: d41d8cd98f00b204e9800998ecf8427e
EXPECTED MD5SUM: 52c65f4f1e8024a78743ac0972395c56
 Computed MD5SUM does not match to the expected one.
 ```
de.NBI recently performed an upgrade, changed its certificates and "Lets encrypt" changed its root certificate, so I thought the BusyBox wget version inside this container was just to old to recognize our new certificates. Newer images worked but also failed with this error:

```
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Database will be downloaded!
Connecting to openstack.cebitec.uni-bielefeld.de:8080 (129.70.51.3:8080)
saving to 'out.zip'
out.zip               27% |********                        | 86.0M  0:00:02 ETA
out.zip               71% |**********************          |  226M  0:00:00 ETA
out.zip              100% |********************************|  316M  0:00:00 ETA
'out.zip' saved
MD5SUM: 52c65f4f1e8024a78743ac0972395c56
EXPECTED MD5SUM: 52c65f4f1e8024a78743ac0972395c56
mash: Relink `/usr/local/bin/../lib/./libgfortran.so.5' with `/lib/x86_64-linux-gnu/librt.so.1' for IFUNC symbol `clock_gettime'
Sketching test2_bin.32_plasmids_filtered.fasta.gz...
Writing to query.msh...
mash: Relink `/usr/local/bin/../lib/./libgfortran.so.5' with `/lib/x86_64-linux-gnu/librt.so.1' for IFUNC symbol `clock_gettime'
chown: unrecognized option '--from'
BusyBox v1.32.1 (2021-04-13 11:15:36 UTC) multi-call binary.

Usage: chown [-RhLHPcvf]... USER[:[GRP]] FILE...

Change the owner and/or group of each FILE to USER and/or GRP

        -R      Recurse
        -h      Affect symlinks instead of symlink targets
        -L      Traverse all symlinks to directories
        -H      Traverse symlinks on command line only
        -P      Don't traverse symlinks (default)
        -c      List changed files
        -v      List all files
        -f      Hide errors
ls: cannot access '.command.sh': No such file or directory
ls: cannot access '.command.log': No such file or directory
```

Again a BusyBox error, the chown version does not support Nextflows after Docker user mapping attempt. The strange part is that this is a known behavior: https://github.com/nextflow-io/nextflow/issues/755 
And it is old. So it should have never worked in the first place.

The root of the problem seams to be, that Nextflow has changed its error handling/recognition. Running the Plsdb process with an old Nextflow version produces this:

```
nxf-scratch-dir bibigrid-worker-2-2-hhmodi5mtqqhxx3:/vol/scratch/nxf.zv3ALjl0tT
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Database will be downloaded!
Connecting to openstack.cebitec.uni-bielefeld.de:8080 (129.70.51.3:8080)
wget: note: TLS certificate validation not implemented
saving to 'out.zip'
out.zip                7% |**                              | 22.2M  0:00:13 ETA
out.zip               15% |****                            | 48.9M  0:00:10 ETA
out.zip               24% |*******                         | 76.1M  0:00:09 ETA
out.zip               32% |**********                      |  102M  0:00:08 ETA
out.zip               40% |*************                   |  129M  0:00:07 ETA
out.zip               49% |***************                 |  156M  0:00:06 ETA
out.zip               57% |******************              |  183M  0:00:05 ETA
out.zip               66% |*********************           |  210M  0:00:04 ETA
out.zip               74% |***********************         |  236M  0:00:03 ETA
out.zip               83% |**************************      |  263M  0:00:02 ETA
out.zip               91% |*****************************   |  288M  0:00:01 ETA
out.zip               98% |******************************* |  313M  0:00:00 ETA
out.zip              100% |********************************|  316M  0:00:00 ETA
'out.zip' saved
MD5SUM: 52c65f4f1e8024a78743ac0972395c56
EXPECTED MD5SUM: 52c65f4f1e8024a78743ac0972395c56
Sketching anafun-PB25-11_plasmid_assembly_plasmids_filtered.fasta.gz...
Writing to query.msh...
chown: unrecognized option '--from'
BusyBox v1.32.1 (2021-03-22 16:41:38 UTC) multi-call binary.

Usage: chown [-RhLHPcvf]... USER[:[GRP]] FILE...

Change the owner and/or group of each FILE to USER and/or GRP

        -R      Recurse
        -h      Affect symlinks instead of symlink targets
        -L      Traverse all symlinks to directories
        -H      Traverse symlinks on command line only
        -P      Don't traverse symlinks (default)
        -c      List changed files
        -v      List all files
        -f      Hide errors
ls: cannot access '.command.sh': No such file or directory
ls: cannot access '.command.log': No such file or directory
```
But everything is marked with exitcode 0, so no error in the CI.

Running the same code with a newer Nextflow version kills the script after the first BusyBox TLS error, at the latest after the chown error if the database already exists. The reason is that all this is now recognized as exitcode 127 (command not found) as it should. 

To sum it up, we need to forego BusyBox images as they are not compatible with Nextflow or put in other words, work, but are producing errors nowadays and we can not really trust Nextflow exitcodes.
